### PR TITLE
Add 2024 GC Election docs

### PR DESCRIPTION
**Pull request by danielgblanco**
Fixes https://github.com/open-telemetry/community/issues/2303

I will keep it as draft until https://github.com/open-telemetry/community/issues/2299 is done and I can add the election ID. This will have to be merged before https://github.com/open-telemetry/community/issues/2304 is done and the issue created. Then, I'll add another PR to include the link to the issue.

### CODE REVIEW

File: elections/2024/governance-committee-election.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+- [Morgan McLean](https://github.com/mtwo), Splunk, until October 2024
+- [Severin Neumann](https://github.com/svrnm), Cisco, until October 2025
+- [Ted Young](https://github.com/tedsuo), Lightstep, until October 2025
+- [Trask Stalnaker](https://github.com/trask), Microsoft, until October 2024
+
+Four people must be elected in this election, each with two-year terms. Note that four current Governance Committee members end terms as part of this election cycle, though they are welcome to run again for one of the four vacancies.
+
+# Voting process
+
+Anyone can track the 2024 election process via [this GitHub issue](https://github.com/open-telemetry/community/issues/{issue_number}). This year's election committee consists of governance committee members Austin Parker and Daniel Gomez Blanco. None of whom are up for reelection this year. We strive for transparency in the election process and hold the community's interests as our priority. In particular, we will ensure that all documents and assets related to the 2024 election process are public, and we will attempt to record and distribute notes for any meetings held as part of this process.
```
Comment (arminru) [inline elections/2024/governance-committee-election.md:43]:
```suggestion
Anyone can track the 2024 election process via [this GitHub issue](https://github.com/open-telemetry/community/issues/{issue_number}). This year's election committee consists of governance committee members Austin Parker, Daniel Gomez Blanco, and Juraci Paixão Kröhling. None of whom are up for reelection this year.
We strive for transparency in the election process and hold the community's interests as our priority. In particular, we will ensure that all documents and assets related to the 2024 election process are public, and we will attempt to record and distribute notes for any meetings held as part of this process.
```

per https://github.com/open-telemetry/community/issues/2300#issuecomment-2312053496

Comment (danielgblanco) [reply elections/2024/governance-committee-election.md]: Thanks!

File: elections/2024/governance-committee-election.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+* GitHub handle
+* List either:
+  * Your contributions to OpenTelemetry that make you eligible to vote, or
+  * Your non-code contributions per the [Members of Standing section of the charter](../../governance-charter.md#members-of-standing): particularly, your relationship to a well-known project that's taken a hard dependency on OpenTelemetry, or your involvement with a well-known organization's adoption of OpenTelemetry as an end-user.
+
+All exemption forms are private. Only the current governance and election committee members will have access to this information. The governance committee will discard the lists one month after the election.
+
+# Vote
+
+Everyone with voting rights may log into [Helios Voting](https://vote.heliosvoting.org/helios/elections/176e7ca8-647d-11ef-9b9a-2a30e2a223da/view) using their GitHub account. Voting will be approval voting, where each voter may select up to five candidates. The five candidates with the most votes win the election.
```
Comment (jpkrohling) [inline elections/2024/governance-committee-election.md:83]:
```suggestion
Everyone with voting rights may log into [Helios Voting](https://vote.heliosvoting.org/helios/elections/176e7ca8-647d-11ef-9b9a-2a30e2a223da/view) using their GitHub account. Voting will be approval voting, where each voter may select up to four candidates. The four candidates with the most votes win the election.
```

File: elections/2024/governance-committee-election.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+One of two options will be available in the form to prove eligibility:
+
+* GitHub handle
+* List either:
+  * Your contributions to OpenTelemetry that make you eligible to vote, or
+  * Your non-code contributions per the [Members of Standing section of the charter](../../governance-charter.md#members-of-standing): particularly, your relationship to a well-known project that's taken a hard dependency on OpenTelemetry, or your involvement with a well-known organization's adoption of OpenTelemetry as an end-user.
+
+All exemption forms are private. Only the current governance and election committee members will have access to this information. The governance committee will discard the lists one month after the election.
```
Comment (jpkrohling) [inline elections/2024/governance-committee-election.md:79]: We might need to add this to the tracker. We've forgotten to do this in the past.

File: elections/2024/governance-committee-election.md (hunk 2 above)
Comment (jpkrohling) [reply elections/2024/governance-committee-election.md:83]: Perhaps we should add a notice to people with uppercase letters on their github handle and point them to this bug? https://github.com/benadida/helios-server/issues/362

They should get in touch with the election committee to request them being added manually to the voter roll.

File: elections/2024/governance-committee-election.md (hunk 3 above)
Comment (danielgblanco) [reply elections/2024/governance-committee-election.md:79]: Thanks. I'll add it.

**Review by jpkrohling — APPROVED** [review]

File: elections/2024/governance-committee-election.md (hunk 3 above)
Comment (danielgblanco) [reply elections/2024/governance-committee-election.md:79]: Done https://github.com/open-telemetry/community/issues/2318

File: elections/2024/governance-committee-election.md (hunk 2 above)
Comment (danielgblanco) [reply elections/2024/governance-committee-election.md]: Added in https://github.com/open-telemetry/community/pull/2315/commits/2d1a779e48608bc6726d1d8a66db615022a5ee4d

**Comment by danielgblanco** [comment]
It's effectively the same document as last year, but I've added a reference to the Helios Issue in https://github.com/open-telemetry/community/pull/2315/commits/2d1a779e48608bc6726d1d8a66db615022a5ee4d

**Comment by danielgblanco** [comment]
cc @open-telemetry/otel-elections

**Review by jpkrohling — APPROVED** [review]

**Review by austinlparker — APPROVED** [review]

**Pull request merged**